### PR TITLE
[53586][feature] detail screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 53586: Detail Meal Screen](https://edu-redmine.sun-asterisk.vn/issues/53586)
- [#Task 53590: Detail Ingredient Screen](https://edu-redmine.sun-asterisk.vn/issues/53590)
## WHAT
- Detail meal screen
- Detail Ingredien screen
- Search ingredient
## Evidence (Screenshot or Video)
627003768f2c4872113d
423bb7733b29fc77a538
cabddc6351399667cf28

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/themeal/data/model/Ingredient.kt [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 
+import android.os.Parcelable
 import androidx.recyclerview.widget.DiffUtil
 import com.google.gson.annotations.SerializedName
+import kotlinx.parcelize.Parcelize
 
+@Parcelize
 data class Ingredient(
     @SerializedName("idIngredient")
     val id: String,
```
Comment (ghost) [inline app/src/main/java/com/example/themeal/data/model/Ingredient.kt]: checknull

File: app/src/main/java/com/example/themeal/data/model/MealCollapse.kt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package com.example.themeal.data.model
 
 import android.os.Parcelable
+import android.util.Log
```
Comment (ghost) [inline app/src/main/java/com/example/themeal/data/model/MealCollapse.kt]: remove log

File: app/src/main/java/com/example/themeal/ui/home/HomeFragment.kt [hunk 3]
```diff
@@ -2,6 +2,7 @@ package com.example.themeal.ui.home
 
 import android.content.Intent
 import android.os.Bundle
+import android.util.Log
```
Comment (ghost) [inline app/src/main/java/com/example/themeal/ui/home/HomeFragment.kt]: remove log

File: app/src/main/java/com/example/themeal/ui/home/adapter/MealAdapter.kt [hunk 4]
```diff
@@ -1,16 +1,21 @@
 package com.example.themeal.ui.home.adapter
 
+import android.util.Log
```
Comment (ghost) [inline app/src/main/java/com/example/themeal/ui/home/adapter/MealAdapter.kt]: .

File: app/src/main/java/com/example/themeal/ui/search/SearchActivity.kt [hunk 5]
```diff
@@ -1,28 +1,61 @@
 package com.example.themeal.ui.search
 
 import android.os.Bundle
+import android.util.Log
```
Comment (ghost) [inline app/src/main/java/com/example/themeal/ui/search/SearchActivity.kt]: .

**Review by ndt0409 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: app/src/main/java/com/example/themeal/custom/ReadMoreTextView.kt [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        showLine = line
+    }
+
+    fun setReadMoreText(readMoreText: String) {
+        this.readMoreText = readMoreText
+    }
+
+    fun setHideText(hideText: String) {
+        this.hideText = hideText
+    }
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/themeal/custom/ReadMoreTextView.kt]: set vao attribute set

**Pull request merged**